src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,164 @@
+import {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
+import {
+  createTodoAPI, deleteTodoAPI, getTodosAPI,
+} from '../../api/todos';
+import { useTodos } from '../hooks/useTodos';
+
+import { Header } from '../Header';
+import { TodoList } from '../TodoList';
+import { Footer } from '../Footer';
+import { Notification } from '../Notification';
+
+import { Todo } from '../../types/Todo';
+import { ShowError } from '../../types/ShowErrors';
+import { ShowTodos } from '../../types/StatusTodo';
+import { USER_ID } from '../../types/USER_ID';
+
+export const TodoApp = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodos, setSelectedTodos] = useState<ShowTodos>(ShowTodos.All);
+  const [title, setTitle] = useState<string | null>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [error, setError] = useState<ShowError | null>(null);
+  const hideError = useCallback(
+    () => setError(null),
+    [],
+  );
+
+  useEffect(() => {
+    const timeoutId = setTimeout(() => {
+      setError(null);
+    }, 3000);
+
+    return () => clearTimeout(timeoutId);
+  }, [error]);
+
+  useEffect(() => {
+    setError(null);
+
+    getTodosAPI()
+      .then(newTodos => setTodos(newTodos))
+      .catch(() => setError(ShowError.fetchTodos));
+  }, []);
+
+  const handleSelectedTodos = useCallback(
+    (event: React.MouseEvent<HTMLAnchorElement>) => {
+      switch (event.currentTarget.textContent) {
+        case 'Active':
+          setSelectedTodos(ShowTodos.Active);
+          break;
+
+        case 'Completed':
```

Comment:
create an enum for todoStatuses like 'Active, completed' etc.

Reply:
Done